conda-envs/windows-environment-dev-py38.yml

```diff
@@ -0,0 +1,37 @@
+name: pymc3-dev-py38
```

Comment:
Is there any particular reason the window dev environment needs to be different? If so add a comment up top so whoever reads this next is aware what those are. 

I havet used windows in many years now so even I dont know, which is where the comment will help

Reply:
The `m2w64-toolchain` dependency is Windows-only, but conda doesn't support OS-conditional dependencies in these files..

I figured it was easier to maintain & diagnose two separate files than to find the set of dependencies that magically makes the conda dependency resolver find the correct result on all platforms.

Reply:
I would add this explanation to the file

Reply:
Deferred to #4786